### PR TITLE
Account Details: Update, Search Test cases

### PR DESCRIPTION
**Pull request by ktnarendra-egov**

### CODE REVIEW

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 1]
```diff
@@ -0,0 +1,128 @@
+Feature: Business Services - HRMS
+
+        Background:
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:3]: Please fix the alignment

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    * def boundary = tenantId
+    * def department = mdmsStatecommonMasters.Department[0].code
+    * def designation = mdmsStatecommonMasters.Designation[0].code
+    * def isCurrentAssignment = true
+    * def fromDate = today
+    * def dateOfAppointment = today
+    * def toDate = null
+    * call read('../../business-services/pretest/egovHrmsPretest.feature@searchEmployeeSuccessfullyWithoutEmployeeCodes')
+    #* call read('../../municipal-services/tests/PropertyService.feature')
+    #* print isCorrespondenceAddress
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:30]: Remove commented lines from the test if it is not required.

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    Scenario: "Use Counter Employee:Validate Property Create and validate doc verifier"
+# Use below qa.yaml details
+# host: https://qa.digit.org/
+# stateCode: pb
+# cityCode: amritsar
+# # Super User credentials for login(need to create user manually)
+# superUser:
+#   userName: EMP-107-000878
+#   password: Password@2
+#   type: EMPLOYEE
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:106]: Remove unnecessary lines from the script.

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    # Create a property
+    * call read('../../municipal-services/pretests/propertyServicesPretest.feature@errorInCreateProperty')
+    # Validate response body
+    * match propertyServiceResponseBody.Properties[0].id == "#present"
+    * match propertyServiceResponseBody.Properties[0].propertyId == "#present"
+    * match propertyServiceResponseBody.Properties[0].status == "INWORKFLOW"
+    * match propertyServiceResponseBody.Properties[0].tenantId == tenantId
+    # * match propertyServiceResponseBody.Errors[0].message == commonConstants.errorMessages.authorizedError
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:114]: Remove the commented validation.

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    Scenario: Create Property invalid tenantId
+    # Set request payload variable values
+    * def tenantId = 'pb.jalandhar'
+    # Create a property
+    * call read('../../municipal-services/pretests/propertyServicesPretest.feature@errorInCreateProperty')
+    # Validate response body
+    * print createPropertyRequest
+    * print propertyServiceResponseBody 
+    * match propertyServiceResponseBody.Errors[0].message == commonConstants.errorMessages.authorizedError
+
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:127]: Please remove the additional spaces as well.

File: src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature [hunk 6]
```diff
@@ -37,4 +37,100 @@ Background:
...
+    }
+    """ 
+    Given url accountDetailTypesSearch
+    And params accountSearchParam
+    And request accountDetailTypesSearchPayload
+    When method post
+    Then status 403
+    And def accountDetailTypesSearchResponseHeader = responseHeaders
+    And def accountDetailTypesSearchResponseBody = response
+
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature:121]: Please try to merge all `negative` pretests in one single pretest scenario, if the steps are similar for all. To validate the range of error status code, please use the below line.
```
assert responseStatus >=400 && responseStatus <=403 
```

**Review by krishnendu-eGov — CHANGES_REQUESTED** [review]
The PR review is done from my side. A couple of comments added. Please look into those and address them accordingly. Also, please add proper documentation for each test step.

File: src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature (hunk 6 above)
Comment (ktnarendra-egov) [reply src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature]: Review comments addressed. Please process.

**Comment by ktnarendra-egov** [comment]
All review comments are addressed, please process.

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 7]
```diff
@@ -0,0 +1,109 @@
+Feature: Business Services - HRMS
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:1]: Invalid Feature name. Please fix it. This is for `zuul` not for `HRMS`

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        * def addressProofDocumentType1 = mdmsStatePropertyTax.Documents[1].dropdownData[0].code
+        * def addressProofDocumentType2 = mdmsStatePropertyTax.Documents[2].dropdownData[0].code
+        * def addressProofDocumentType3 = mdmsStatePropertyTax.Documents[3].dropdownData[0].code
+        * def addressProofDocumentType4 = mdmsStatePropertyTax.Documents[4].dropdownData[0].code
+        * def creationReason = commonConstants.parameters.creationReason
+        * def businessService = mdmsStatePropertyTax.PTWorkflow[1].businessService
+        * def propertyServicesConstants = read('../../municipal-services/constants/propertyServices.yaml')
+    
+    @zuul_01 @zuul @regression @positive @smoke
+        Scenario: Test to create a employee
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:71]: Invalid `Scenario` name, this should be `Test case description` as per the test case sheet.

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    * call read('../../business-services/pretest/egovHrmsPretest.feature@createEmployeeSuccessfully')
+    * def code = hrmsResponseBody.Employees[0].user.userName
+    * assert hrmsResponseBody.ResponseInfo.status == commonConstants.expectedStatus.success
+    * assert hrmsResponseBody.Employees[0].user.name == name
+    * assert hrmsResponseBody.Employees[0].user.mobileNumber == mobileNumber
+
+    @zuul_02 @zuul  @positive @regression  @municipalServices
+    Scenario: Create Property with valid payload
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:79]: Please fix the `Scenario` names for all of the tests. This should be a test case description as per the test case sheet.

File: src/test/java/com/egov/core-services/tests/zuul.feature [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    # Create a property
+    * call read('../../municipal-services/pretests/propertyServicesPretest.feature@createPropertySuccessfully')
+    # Validate response body
+    * match propertyServiceResponseBody.Properties[0].id == "#present"
+    * match propertyServiceResponseBody.Properties[0].propertyId == "#present"
+    * match propertyServiceResponseBody.Properties[0].status == "INWORKFLOW"
+    * match propertyServiceResponseBody.Properties[0].tenantId == tenantId
+
+    @zuul_05  @zuul @regression @negative
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/core-services/tests/zuul.feature:88]: please add two more tags here. i.e. `@coreServices`, and also specify whether this is a `positive` or `negative test. Like, `@positive` or `@nagetive`.

File: src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature [hunk 11]
```diff
@@ -1,40 +1,121 @@
...
+      """
+      {
+      tenantId: '#(tenantId)'
+      }
+      """ 
+      Given url accountDetailTypesSearch
+      And params accountSearchParam
+      And request accountDetailTypesSearchPayload
+      When method post
+    # Then status 403
```
Comment (krishnendu-eGov) [inline src/test/java/com/egov/business-services/pretest/egfMasterAccountDetailsTypesPreTest.feature:118]: Remove this commented line

**Review by krishnendu-eGov — CHANGES_REQUESTED** [review]
Few more review comments provide. Please address. These are minor changes but need to be fixed.

**Comment by ktnarendra-egov** [comment]
All reviews addressed : zuul tagging and Acc details

**Review by krishnendu-eGov — APPROVED** [review]

**Pull request merged**